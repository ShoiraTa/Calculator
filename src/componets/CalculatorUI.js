/* eslint-disable react/prefer-stateless-function */
/* eslint-disable no-useless-constructor */

import './calculatorUI.css';
import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: '0', next: null, operation: null });

  const handleClick = (e) => {
    setState(calculate(state, e.target.innerText));
  };

  return (
    <>

      <div className="calculator">
        <h1>Let&apos;s do some Math</h1>
        <div className="calculator-container">
          <div className="result">
            {state.total}
            {state.operation}
            {state.next}
          </div>
          <div className="calculator-oprtaitions">
            <button type="button" onClick={handleClick} className="bg-gray single-cell ">AC</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell ">%</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell ">+/-</button>
            <button type="button" onClick={handleClick} className="bg-orange single-cell">&divide;</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell">7</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell ">8</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell ">9</button>
            <button type="button" onClick={handleClick} className="bg-orange single-cell">x</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell ">4</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell">5</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell">6</button>
            <button type="button" onClick={handleClick} className="bg-orange single-cell">-</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell">1</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell">2</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell">3</button>
            <button type="button" onClick={handleClick} className="bg-orange single-cell">+</button>
            <button type="button" onClick={handleClick} className="bg-gray zero single-cell">0</button>
            <button type="button" onClick={handleClick} className="bg-gray single-cell">.</button>
            <button type="button" onClick={handleClick} className="bg-orange single-cell">=</button>
          </div>
        </div>
      </div>
    </>
  );
};

export default Calculator;
