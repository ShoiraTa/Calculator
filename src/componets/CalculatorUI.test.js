import React from 'react';
import renderer from 'react-test-renderer';
import CalculatorUI from './CalculatorUI';

test('matches home snapshot', () => {
  const tree = renderer.create(<CalculatorUI />).toJSON();
  expect(tree).toMatchSnapshot();
});
