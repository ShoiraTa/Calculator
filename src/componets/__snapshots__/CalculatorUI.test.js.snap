// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches home snapshot 1`] = `
<div
  className="calculator"
>
  <h1>
    Let's do some Math
  </h1>
  <div
    className="calculator-container"
  >
    <div
      className="result"
    >
      0
    </div>
    <div
      className="calculator-oprtaitions"
    >
      <button
        className="bg-gray single-cell "
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="bg-gray single-cell "
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="bg-gray single-cell "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="bg-orange single-cell"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="bg-gray single-cell"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="bg-gray single-cell "
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="bg-gray single-cell "
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="bg-orange single-cell"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="bg-gray single-cell "
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="bg-gray single-cell"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="bg-gray single-cell"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="bg-orange single-cell"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="bg-gray single-cell"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="bg-gray single-cell"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="bg-gray single-cell"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="bg-orange single-cell"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="bg-gray zero single-cell"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="bg-gray single-cell"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="bg-orange single-cell"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
